### DISCUSSION TITLE
Tracking Protection no Firefox

### DISCUSSION THREAD

**Discussion by celsomtrindade** [frontendbr/forum#711] [Perguntas e Respostas]
Boa noite pessoal, tudo certo?

Seguinte, estou trabalhando em uma aplicação (nova) e comecei a receber críticas sobre algumas coisas não funcionando corretamente no Firefox. Quando fui pesquisar, notei que isso se deve a um problema com o [tracking protection](https://developer.mozilla.org/en-US/Firefox/Privacy/Tracking_Protection) que impede o carregamento de determinados scripts na página.

Neste caso em específico, o problema se refere ao Login via Facebook e algumas funcionalidades do GoogleAnalytics.

Isso acontece apenas no Firefox, por exemplo, ao chamar a função `FB.getLoginStatus()` nada acontece, nenhum erro, nenhuma função, etc.

Ao chamar `FB.login()` a caixa de diálogo até é exibida, porém ela nunca é resolvida, ou seja, quando o usuário confima/cancela, o popup fica em branco mas nunca fecha nem resolve.

Alguém já passou por isso/sabe como posso resolver?

> Obs.: Sei que posso resolver localmente desabilitando a flag do navegador, porém isso não vai evitar de um usuário do meu site passar pelo mesmo problema e apenas pensar que o site em si que está quebrado e não que é uma funcionalidade do firefox.

**Comment by felipefialho** [comment]
Alguém já viu isso? @frontendbr/admins

**Comment by willianjusten** [comment]
Bom, pelo que entendi pelo próprio link que você postou, o Firefox só bloqueia esses acessos externos quando em aba anônima, exatamente para não enviar/salvar dados, que é o procedimento padrão.

Outra coisa que você precisa fazer é verificar se o objeto de fato existe para de fazer chamar o seu método, também explicado no post. Eu fui até verificar em sistemas onde uso Analytics e Fb, ambos funcionam de boas... Já verificou se no Firefox você não está usando como anônimo ou de alguma forma "bloqueando" tudo? Tenta em outro computador ou até mesmo um browserstack para ter certeza que não é seu ambiente te enganando...

**Comment by eduardojmatos** [comment]
@celsomtrindade cara, isso acontece muito no Brave. Trackings de informações, eventos, etc. tudo é bloqueado. Tenta rodar no browserstack em algumas versões diferentes do Firefox pra testar.

**Comment by munizart** [comment]
@eduardojmatos outra forma de verificar se o usuario consente com o tracking é o DNT
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/DNT

**Comment by celsomtrindade** [comment]
O engraçado é que por mais que o script fosse bloqueado, algumas funções eram chamadas e retornavam corretamente (como o caso do FB.api), algumas eram chamadas mas funcionavam parcialmente e algumas simplesmente não funcionavam, ou seja, o método FB (como questionado pelo @willianjusten ) de fato existia no momento que precisava ser chamado.

Pesquisei bastante e não consegui encontrar nenhuma solução clara. Isso foi testado no Firefox, tanto em anônimo como navegação normal, o comportamento era sempre o mesmo. Testei em algumas versões do Firefox e obtive os dois resultados, funcionando 100% e apresentando o erro descrito na pergunta. Quando voltei para a versão mais recente, voltou a funcionar corretamente e, desde então, não tenho mais visto o problema. Engraçado... Pode ter sido algo na própria instalação do Firefox que tenha apresentado esse erro.

Lendo mais a respeito, encontrei alguns artigos falando sobre a presença deste erro (ou semelhantes) quando o navegador possui `adblocker`. No meu caso eu não tenho instalado isso no Firefox, porém tenho instalado no Chrome, onde funciona perfeitamente, mesmo com `adblocker` ligado.

Enfim.. Não sei dizer ao certo o que causou esse erro nem o que eu fiz para resolver. Mas, aparentemente, resolveu.
